IotWebConf updated to V3.x.x from V2.3.3

markvader opened the pull request:
As per pull request from @GerdZanker https://github.com/pinae/WiFi-Poti-Fan/pull/1 this is no longer compiling with latest version of IotWebConf.  I have updated to V3.2.0
See migration guide for more details.
https://github.com/prampec/IotWebConf/blob/master/doc/MigrationGuide-v3.0.0.md

markvader committed:
IotWebConf updated to V3.x.x from V2.3.3

As per pull request from @GerdZanker https://github.com/pinae/WiFi-Poti-Fan/pull/1 this is no longer compiling with latest version of IotWebConf.  I have updated to V3.2.0
See migration guide for more details.
https://github.com/prampec/IotWebConf/blob/master/doc/MigrationGuide-v3.0.0.md

GerdZanker commented:
I get compiler errors in my setup, using latest Espressif 8266 v3.2.0 platform like and prampec/IotWebConf v3.2.0

    src\configHelpers.cpp:20:1: error: 'HTTPUpdateServer' does not name a type
    ...
    src\configHelpers.cpp:115:52: error: 'PWMRANGE' was not declared in this scope

GerdZanker commented:
> I get compiler errors in my setup... 

With the code from master branch and IotWebConf v2.x the compiler still complains about the not declared PWMRANGE.
Something must be broken in my setup.

markvader committed:
additional changes needed to updateServer for IotWebConf v3.x.x

Added ESP8266HTTPUpdateServer.h dependency.

https://github.com/prampec/IotWebConf/blob/b95c15aaf6887ccd574436e4cfb619a2035622a9/doc/MigrationGuide-v3.0.0.md#updateserver-changes

markvader commented:
The HTTPUpdateServer should be fixed, it was an additional step in the IotWebConf Migration guide
https://github.com/prampec/IotWebConf/blob/master/doc/MigrationGuide-v3.0.0.md#updateserver-changes

markvader committed:
declare PWMRANGE variable

markvader committed:
add PWMRANGE (extern) variable

markvader commented:
For some reason it would compile for me with Espressif ESP8266 2.7.4 but not 3.0.2

I have added the PWMRANGE variable so now compiles on latest Espressif ESP8266 platform (3.0.2)

Let me know how it goes.

GerdZanker commented:
Yes, with the old "2.x" vesions I had also no issues.

With your latest code I get now a linker errors ``undefined reference to `_ZTVN10iotwebconf17WebRequestWrapperE'`` and ``undefined reference to `_ZTVN10iotwebconf16WebServerWrapperE'``. The compiler doesn't complain anymore about problems.
I use the latest Espressif ESP8266 platform 3.*2.0*
I have no good glue why linking fails- The only different I was able to find is the different #define if ESP8266 is used:

    #ifdef ESP8266
    # ifndef WebServer
    #  define WebServer ESP8266WebServer
    # endif
    #endif

markvader committed:
change build type to release

markvader committed:
bump config version to v0.2.2

markvader commented:
Can you check does it build now.  I previously was using ArduinoIDE (so was therefore not using Platformio file).  I encountered the same errors as yourself when i first tried Platformio but upon switching to release build rather than debug build I didn't have issues.

GerdZanker commented:
Release Build is compiling and linking successfully - all is perfect now.
Thanks a lot @markvader for solving all the issues.

markvader commented:
@pinae When it suits, do you want to test this?  Thanks

pinae commented:
I tried to test this patch but failed.

1. The code does not compile with PWMRANGE set as a local variable. This is a `#define` in `Arduino.h` and its value is usually 1023. After I removed the two lines it did compile.
2. I have two devices running the code. One is a air filter which I use a lot since the pandemic is not over yet. The electronics is buried inside the case and I don't want to mess with it at the moment. The other device is a fan which cools my camera. I flashed this one and PlatforIO was able to upload the code. The debug output looks exactly as it should but it does not create an AP for the configuration network. Hardware-wise that's not a big deal for me as I use this fan as a offline device. However I can't verify if the code is working as expected. In my case the test was not successful but cause of the failure could be my hardware.

This is a dissatisfying situation. I am very thankful for the code update and I would like to integrate the update. The next step for me would be to build another board for the test but I need to finish several other projects before that. I hope to build the test board later this year. Sorry for the delay...

markvader committed:
Fix to PWMRANGE

The analogWrite-Range is no longer 1023 by default. It has been changed to 255.
"NOTE: The default analogWrite range was 1023 in releases before 3.0, but this lead to incompatibility with external libraries which depended on the Arduino core default of 256. Existing applications which rely on the prior 1023 value may add a call to analogWriteRange(1023) to their setup() routine to return to their old behavior." 
https://arduino-esp8266.readthedocs.io/en/latest/reference.html?highlight=analogwrite#analog-output

markvader committed:
Use IRAM_ATTR in place of ICACHE_RAM_ATTR

https://github.com/esp8266/Arduino/pull/7921

markvader committed:
Provide legend to MQTT section in AP configuration

markvader commented:
I haven't forgot about this project.  The last fixes I made sort the PWMRANGE among others, I do agree that there is an issue remaining that still needs sorted. 
I believe it is something else in the loop causing the issue currently as when I comment out all the lines except 
```
void loop() {
  logger.printAllWithSerial();
  iotWebConf.doLoop();
  loopWifiChecks();
```
It loads the access point for setup as expected, I will get back to it and debug further
